File: javascript/fibonacci.js
function fibonacci(num) {
  var f = 0
  const arr = [0, 1]
  for (var i = 1; i < num; i++) {
    f = arr[i]
    arr.push(f + arr[i - 1])
    f++
  }
  let last = arr[num];
  return(last)
}

if (require.main === module) {

  console.log("Expecting: 0");
  console.log("=>", fibonacci(0));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(2));

  console.log("");

  console.log("Expecting: 55");
  console.log("=>", fibonacci(10));
}

module.exports = fibonacci;

File: ruby/fibonacci.rb
def fibonacci(num)

  $f = 0
  $arr = [0, 1]
  $i = 1
  while $i < num do
    $f = $arr[$i]
    $arr << $f + $arr[$i - 1]
    $f +=1
    $i +=1
  end
  # last = arr[num];
  # last
  $arr[num]

end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts "=>", fibonacci(0)

  puts

  puts "Expecting: 1"
  puts "=>", fibonacci(2)

  puts

  puts "Expecting: 55"
  puts "=>", fibonacci(10)

end
